src/UserManager.php

```diff
@@ -161,27 +167,31 @@ public function loadManagedUserIds(): array {
         }
       }
 
-      $userIds = $query->execute();
+      $managedUserIds = $query->execute();
     }
 
-    return $userIds;
+    return $managedUserIds;
   }
 
   /**
    * {@inheritdoc}
    */
   public function collectUsersForDeletionList(): void {
-    $userIds = $this->loadManagedUserIds();
-    $this->cacheUserIdsForDeletion($userIds);
-    $this->logger->info($this->formatPlural(
-      count($userIds),
-      '1 user marked for deletion.',
-      '@count users marked for deletion.'
-    ));
+    $managedUserIds = $this->loadManagedUserIds();
+    if (0 !== count($managedUserIds)) {
+      $this->userIds = $managedUserIds;
+      $this->logger->info($this->formatPlural(
+        count($this->userIds),
+        '1 user marked for deletion.',
+        '@count users marked for deletion.'
+      ));
+    }
```

Comment:
I think we should skip the check:

```suggestion
    $this->userIds = $managedUserIds;
    $this->logger->info($this->formatPlural(
      count($this->userIds),
      '1 user marked for deletion.',
      '@count users marked for deletion.'
    ));
```

If we don't reset the list (even when it's empty), we can risk reusing a non-empty list and delete too many users. It's theoretical and unlikely, but I still think we always user the actual list of managed user IDs.

Reply:
Why skip the count?

Reply:
Because we must always process the list even when it's empty. It will probably never be empty, but assume this (contrived) example:

1. We run delta sync and are told to delete user 2
2. User 2 is now deleted
3. We run delta sync again and are told to delete no users (since user 2 is already deleted), but if we only process non-empty lists, we will reuse the list containing user 2